packages/next/client/image.tsx

```diff
@@ -376,7 +381,7 @@ export default function Image({
     unoptimized = true
     isLazy = false
   }
-  if (typeof window !== 'undefined' && loadedImageURLs.has(src)) {
+  if (typeof window !== 'undefined' && ref?.current?.complete) {
```

Comment:
```suggestion
  if (typeof window !== 'undefined' && ref.current?.complete) {
```

`.current` should always exist.